src/Evolution/Systems/Cce/ScriPlusValues.cpp

```diff
@@ -0,0 +1,374 @@
+                  number_of_angular_points);
+
+  const SpinWeighted<ComplexDataVector, 2> dy_j_at_scri;
+  make_const_view(make_not_null(&dy_j_at_scri), get(dy_bondi_j),
+                  (number_of_radial_points - 1) * number_of_angular_points,
+                  number_of_angular_points);
+
+  const SpinWeighted<ComplexDataVector, 1> dy_q_at_scri;
+  make_const_view(make_not_null(&dy_q_at_scri), get(dy_bondi_q),
+                  (number_of_radial_points - 1) * number_of_angular_points,
+                  number_of_angular_points);
+  const SpinWeighted<ComplexDataVector, 0> ethbar_dy_q_at_scri;
+  make_const_view(make_not_null(&ethbar_dy_q_at_scri), get(ethbar_dy_bondi_q),
+                  (number_of_radial_points - 1) * number_of_angular_points,
+                  number_of_angular_points);
+
+  const SpinWeighted<ComplexDataVector, 1> dy_u_at_scri;
+  make_const_view(make_not_null(&dy_u_at_scri), get(dy_bondi_u),
+                  (number_of_radial_points - 1) * number_of_angular_points,
+                  number_of_angular_points);
+  const SpinWeighted<ComplexDataVector, 2> eth_dy_u_at_scri;
+  make_const_view(make_not_null(&eth_dy_u_at_scri), get(eth_dy_bondi_u),
+                  (number_of_radial_points - 1) * number_of_angular_points,
+                  number_of_angular_points);
+
+  const SpinWeighted<ComplexDataVector, 1> dy_dy_u_at_scri;
+  make_const_view(make_not_null(&dy_dy_u_at_scri), get(dy_dy_bondi_u),
+                  (number_of_radial_points - 1) * number_of_angular_points,
+                  number_of_angular_points);
+  const SpinWeighted<ComplexDataVector, 0> ethbar_dy_dy_u_at_scri;
+  make_const_view(make_not_null(&ethbar_dy_dy_u_at_scri),
+                  get(ethbar_dy_dy_bondi_u),
+                  (number_of_radial_points - 1) * number_of_angular_points,
+                  number_of_angular_points);
+
+  const SpinWeighted<ComplexDataVector, 0> exp_2_beta_at_scri;
+  make_const_view(make_not_null(&exp_2_beta_at_scri), get(exp_2_beta),
+                  (number_of_radial_points - 1) * number_of_angular_points,
+                  number_of_angular_points);
+
+  const SpinWeighted<ComplexDataVector, 1> eth_r_divided_by_r_view;
+  make_const_view(make_not_null(&eth_r_divided_by_r_view),
+                  get(eth_r_divided_by_r),
+                  (number_of_radial_points - 1) * number_of_angular_points,
+                  number_of_angular_points);
+
+  get(*psi_2) =
+      -0.5 * get(boundary_r) *
+      (-exp_2_beta_at_scri * (conj(ethbar_dy_q_at_scri) +
+                              eth_r_divided_by_r_view * conj(dy_q_at_scri)) +
+       get(boundary_r) * (ethbar_dy_dy_u_at_scri +
+                          conj(eth_r_divided_by_r_view) * dy_dy_u_at_scri +
+                          conj(ethbar_dy_dy_u_at_scri) +
+                          eth_r_divided_by_r_view * conj(dy_dy_u_at_scri)) +
+       2.0 * get(boundary_r) * dy_j_at_scri *
```

Comment:
if you are making a list of future optimizations, you could factor out the `2.0 * get(boundary_r)`